titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/DefaultV3JobOperations.java

```diff
@@ -570,13 +571,15 @@ public Completable recordTaskPlacement(String taskId, Function<Task, Task> chang
     }
 
     private <E extends JobDescriptor.JobDescriptorExt> Job<E> newJob(JobDescriptor<E> jobDescriptor) {
+        String federatedJobId = jobDescriptor.getAttributes().get(JobAttributes.JOB_ATTRIBUTES_FEDERATED_JOB_ID);
```

Comment:
Should this be removed from the `jobDescriptor` once it is read? My concern is that this gets copied over and we have id conflicts.

Reply:
Thank you. I will create a separate PR for handling this.